core/unit_test/TestBitManipulationBuiltins.hpp

```diff
@@ -118,8 +118,11 @@ void test_bit_manip_countl_zero() {
 }
 
 TEST(TEST_CATEGORY, bit_manip_countl_zero) {
+// FIXME_NVHPC: NVC++-W-0155-Compiler failed to translate accelerator region
+#if !defined(KOKKOS_COMPILER_NVHPC)
```

Comment:
Did you try to restrict further?  It passes with NVC++ when doing CUDA builds so the right thing to do here is probably something like
```C++
#ifdef KOKKOS_ENABLE_OPENACC
if constexpr (!std::is_same_v<TEST_EXECSPACE, Kokkos::Experimental::OpenACC>)
{
#endif
```

Reply:
> Did you try to restrict further? It passes with NVC++ when doing CUDA builds so the right thing to do here is probably something like
> 
> ```c++
> #if defined(KOKKOS_ENABLE_OPENACC) && defined(KOKKOS_COMPILER_NVHPC)
> if constexpr (!std::is_same_v<TEST_EXECSPACE, Kokkos::Experimental::OpenACC>)
> {
> #endif
> ```

You are correct; NVHPC fails only when compiling for the OpenACC backend; the serial backend didn't fail. 
Updated the code as suggested.